internal/model/modelindexer/indexer.go

```diff
@@ -434,6 +393,60 @@ func (i *Indexer) flush(ctx context.Context, bulkIndexer *bulkIndexer) error {
 	return nil
 }
 
+// runActiveIndexer starts a new active indexer which pulls items from the
+// bulkItems channel. The more active indexers there are, the faster events
+// will be pulled out of the queue, but also the more likely it is that the
+// outgoing Elasticsearch bulk requests are flushed due to the idle timer,
+// rather than due to being full.
+func (i *Indexer) runActiveIndexer() {
+	atomic.AddInt64(&i.activeBulkRequests, 1)
+	defer atomic.AddInt64(&i.activeBulkRequests, -1)
+	var closed bool
+	var active *bulkIndexer
+	flushTimer := time.NewTimer(i.config.FlushInterval)
+	if !flushTimer.Stop() {
+		<-flushTimer.C
+	}
+	for !closed {
+		select {
+		case <-flushTimer.C:
+		default:
+			select {
+			case <-flushTimer.C:
+			case event, ok := <-i.bulkItems:
+				if !ok {
+					closed = true
+					break
+				}
+				if active == nil {
+					active = <-i.available
```

Comment:
Should this be a select with `i.closed`, to cause the loop to end immediately when the context passed to Close is cancelled?

Reply:
Isn't that redundant? Given that we close the `i.bulkItems` when the indexer is closing? Maybe I'm missing an obvious condition

Reply:
What I'm wondering is if it's possible for `<-available` to block if the indexer is closed after we've received from bulkItems. I suppose not, since flushIndexer will put the indexers back into available when `i.closed` is signalled. So never mind :)